### PR TITLE
Rename CanonicalCode to StatusCanonicalCode

### PR DESCRIPTION
**Pull request by cijothomas**
Bit longer name, but the spec says so and other languages have named it exactly as per spec.

https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#statuscanonicalcode

https://github.com/open-telemetry/opentelemetry-python/blob/abdfa12dfd76613bef5cc5e9266439e983074bbc/opentelemetry-api/src/opentelemetry/trace/status.py#L19

### CODE REVIEW

File: src/OpenTelemetry.Api/Trace/Status.cs [hunk 1]
```diff
@@ -158,7 +158,7 @@ internal Status(CanonicalCode canonicalCode, string description = null)
         /// <summary>
         /// Gets the canonical code from this status.
         /// </summary>
-        public CanonicalCode CanonicalCode { get; }
+        public StatusCanonicalCode StatusCanonicalCode { get; }
```
Comment (SergeyKanzhelev) [inline src/OpenTelemetry.Api/Trace/Status.cs:161]: I understand why rename the enum. Should we keep the property to be just CanonicalCode? It is already a part of a Status class.

Comment (SergeyKanzhelev) [reply src/OpenTelemetry.Api/Trace/Status.cs:161]: Same python file you are referring: https://github.com/open-telemetry/opentelemetry-python/blob/abdfa12dfd76613bef5cc5e9266439e983074bbc/opentelemetry-api/src/opentelemetry/trace/status.py#L172-L175

Comment (cijothomas) [reply src/OpenTelemetry.Api/Trace/Status.cs:161]: Agree.

**Comment by cijothomas** [comment]
> ok to rename enum,let's keep the property name

Done.

**Comment by SergeyKanzhelev** [comment]
Now there is a conflict becasue of the project rename.

**Comment by cijothomas** [comment]
> Now there is a conflict becasue of the project rename.

done!

**Pull request merged**